test/lib/pgBackRestTest/Module/Manifest/ManifestAllTest.pm

```diff
@@ -0,0 +1,1088 @@
+            false, 'boolGet() - false');
+
+        # Set and get a boolean value
+        #---------------------------------------------------------------------------------------------------------------------------
+        $oManifest->boolSet(MANIFEST_SECTION_TARGET_FILE, BOGUS, MANIFEST_SUBKEY_CHECKSUM_PAGE, true);
+        $self->testResult(sub {$oManifest->boolGet(MANIFEST_SECTION_TARGET_FILE, BOGUS, MANIFEST_SUBKEY_CHECKSUM_PAGE)}, true,
+            'boolGet() - true');
+
+        # Get default boolean value
+        #---------------------------------------------------------------------------------------------------------------------------
+# CSHANG But this doesn't actually set the default value, so should we be allowing this? Why?
+        $self->testResult(sub {$oManifest->boolGet(MANIFEST_SECTION_BACKUP_OPTION, MANIFEST_KEY_COMPRESS, undef, false, true)},
+            true, 'boolGet() - default true');
+
+        # Get but don't return a default boolean value even if there is supposed to be one
+        #---------------------------------------------------------------------------------------------------------------------------
+# CSHANG Is this right? Should the calling routine be passing whether the option has a default or not?
+        $self->testResult(sub {$oManifest->boolGet(MANIFEST_SECTION_BACKUP_OPTION, MANIFEST_KEY_COMPRESS, undef, false, false)},
+            false, 'boolGet() - default false');
+
+        # Get default numeric value
+        #---------------------------------------------------------------------------------------------------------------------------
+# CSHANG But this doesn't actually set the default value, so should we be allowing this? Why?
+        $self->testResult(sub {$oManifest->numericGet(MANIFEST_SECTION_TARGET_FILE, BOGUS, MANIFEST_SUBKEY_SIZE, false, 0)},
+            0, 'numericGet() - default 0');
+
+        # Coverage for $self->SUPER::get("${strSection}:default"
+        #---------------------------------------------------------------------------------------------------------------------------
+# CSHANG This returns undef because required = false - is that good?
+        $self->testResult(sub {$oManifest->get(MANIFEST_SECTION_TARGET_FILE, BOGUS, MANIFEST_SUBKEY_USER, false)},
+            "[undef]", 'get() - default section');
+
+        # Get the correct path for the control file in the DB
+        #---------------------------------------------------------------------------------------------------------------------------
+# CSHANG Should MANIFEST_SUBKEY_TYPE be required for MANIFEST_SUBKEY_PATH? And why is there a MANIFEST_SUBKEY_FILE - shouldn't there be a MANIFEST_VALUE_FILE instead? pg_data/postgresql.conf={"file":"postgresql.conf","path":"../pg_config","type":"link"} is the only time I see it in the mock expect tests and the type is link - in restore, I see it first tests for link and then file - is this the only way "file" is used?
```

Comment:
Do you see any instances where MANIFEST_SUBTYPE_KEY is not being set for MANIFEST_SUBKEY_PATH?  I couldn't find any.

And yes - file links are the only time the "file" type is used.